### PR TITLE
Adding New All-In-One template for Vnet Injection with NAT Gateway

### PR DESCRIPTION
**Pull request by sunainakrishnamoorthy**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Adding files for All in one template with NAT Gateway

### CODE REVIEW

File: 101-databricks-all-in-one-template-for-vnet-injection-with-nat-gateway/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    }
+  },
+  "variables": {
+    "managedResourceGroupName": "[concat('databricks-rg-', parameters('workspaceName'), '-', uniqueString(parameters('workspaceName'), resourceGroup().id))]",
+    "managedResourceGroupId": "[subscriptionResourceId('Microsoft.Resources/resourceGroups', variables('managedResourceGroupName'))]",
+    "nsgId": "[resourceId('Microsoft.Network/networkSecurityGroups', parameters('nsgName'))]",
+    "vnetId": "[resourceId('Microsoft.Network/virtualNetworks', parameters('vnetName'))]"
+  },
+  "resources": [
+    {
```
Comment (bmoore-msft) [inline 101-databricks-all-in-one-template-for-vnet-injection-with-nat-gateway/azuredeploy.json:108]: why is this empty?

File: 101-databricks-all-in-one-template-for-vnet-injection-with-nat-gateway/GettingStarted.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+**Private Subnet Name** Name of the Private Subnet
+**Public Subnet Name** Name of the Public Subnet
+**Location** Location of Data Center
+**Vnet Cidr** Cidr Range of the Vnet
+**Private Subnet Cidr** Cidr Range of the Private Subnet
+**Public Subnet Cidr** Cidr Range of the Public Subnet
+**NAT Gateway Name** Name of the NAT Gateway
+**NAT Gateway Public IP Name** Name of the NAT Gateway Public IP
+
+## Deployment
```
[the remark this answers is not in the document]
Comment (sunainakrishnamoorthy) [reply 101-databricks-all-in-one-template-for-vnet-injection-with-nat-gateway/GettingStarted.md:29]: Removed, and updated the readme.

[the remark this answers is not in the document]
Comment (sunainakrishnamoorthy) [reply 101-databricks-all-in-one-template-for-vnet-injection-with-nat-gateway/GettingStarted.md:29]: Hi Brian, can you please clarify, do I need to remove the entire "Template" section or just the fields that were unique to this template?

[the remark this answers is not in the document]
Comment (bmoore-msft) [reply 101-databricks-all-in-one-template-for-vnet-injection-with-nat-gateway/GettingStarted.md:29]: The ##Deployment section of the file should be removed - after that's remove, what's remaining in this file should be moved to the readme.md (And this file can be removed)

File: 101-databricks-all-in-one-template-for-vnet-injection-with-nat-gateway/azuredeploy.json (hunk 1 above)
Comment (sunainakrishnamoorthy) [reply 101-databricks-all-in-one-template-for-vnet-injection-with-nat-gateway/azuredeploy.json:108]: removed empty properties field

File: 101-databricks-all-in-one-template-for-vnet-injection-with-nat-gateway/GettingStarted.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (sunainakrishnamoorthy) [reply 101-databricks-all-in-one-template-for-vnet-injection-with-nat-gateway/GettingStarted.md]: done

**Pull request merged**